packages/common/src/converter/payload-converter.ts

```diff
@@ -140,3 +149,130 @@ export class BinaryPayloadConverter extends AsyncFacadePayloadConverter {
     return content.data as any;
   }
 }
+
+abstract class ProtobufPayloadConverter extends AsyncFacadePayloadConverter {
+  protected messageClasses: Record<string, ProtobufSerializable> = {};
+
+  constructor(protobufClasses?: Record<string, unknown>) {
+    super();
+
+    if (protobufClasses) {
+      if (typeof protobufClasses !== 'object') {
+        throw new TypeError('protobufClasses must be an object');
+      }
+
+      let count = 0;
+      for (const [name, klass] of Object.entries(protobufClasses)) {
+        if (this.isProtobufMessageClass(klass)) {
+          this.messageClasses[name] = klass;
+          count++;
+        }
+      }
+
+      if (count === 0) {
+        throw new TypeError(
+          'protobufClasses must be an object with values that are classes that have `encode`, `decode`, and `create` static methods'
+        );
+      }
+    }
+  }
+
+  protected isProtobufMessageClass(messageClass: unknown): messageClass is ProtobufSerializable {
+    return (
+      typeof messageClass === 'function' &&
+      typeof (messageClass as unknown as ProtobufSerializable).create === 'function' &&
+      typeof (messageClass as unknown as ProtobufSerializable).encode === 'function' &&
+      typeof (messageClass as unknown as ProtobufSerializable).decode === 'function'
+    );
+  }
+
+  protected isProtobufMessageInstance(value: unknown): value is ProtobufSerializable {
+    if (!value || typeof value !== 'object') {
+      return false;
+    }
+
+    const messageClass = this.messageClasses[value.constructor.name];
```

Comment:
If you have the reverse mapping I mentioned you should be able to lookup by class and get the registered name